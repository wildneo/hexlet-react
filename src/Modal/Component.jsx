import React from 'react';
import Modal from './Modal';

export default class Component extends React.Component {
  constructor(props) {
    super(props);
    this.state = { modal: false };

    this.toggle = () => {
      const { modal } = this.state;
      this.setState({
        modal: !modal,
      });
    };
  }


  render() {
    const { modal } = this.state;

    return (
      <div>
        <button type="button" className="modal-open-button btn btn-danger" onClick={this.toggle}>Open</button>
        <Modal isOpen={modal}>
          <Modal.Header toggle={this.toggle}>Modal title</Modal.Header>
          <Modal.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit
          </Modal.Body>
          <Modal.Footer>
            <button type="button" className="modal-close-button btn btn-default" onClick={this.toggle}>Cancel</button>
          </Modal.Footer>
        </Modal>
      </div>
    );
  }
}
