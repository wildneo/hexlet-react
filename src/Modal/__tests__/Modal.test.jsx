import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Component from '../Component';

Enzyme.configure({ adapter: new Adapter() });

test('Component 1', () => {
  const wrapper = mount(<Component />);
  expect(wrapper.render()).toMatchSnapshot();

  const button = wrapper.find('.modal-open-button');
  button.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  button.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  button.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('.modal-close-button').simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  button.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('.modal-close-button').simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});

test('Close button', () => {
  const wrapper = mount(<Component />);
  const button = wrapper.find('.modal-open-button');
  button.simulate('click');
  wrapper.find('.close').simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});
