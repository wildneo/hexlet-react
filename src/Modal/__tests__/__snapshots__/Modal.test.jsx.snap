// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close button 1`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
    type="button"
  >
    Open
  </button>
  <div
    class="modal"
    style="display: none;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="modal-close-button btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component 1 1`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
    type="button"
  >
    Open
  </button>
  <div
    class="modal"
    style="display: none;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="modal-close-button btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component 1 2`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
    type="button"
  >
    Open
  </button>
  <div
    class="modal fade show"
    style="display: block;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="modal-close-button btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component 1 3`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
    type="button"
  >
    Open
  </button>
  <div
    class="modal"
    style="display: none;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="modal-close-button btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component 1 4`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
    type="button"
  >
    Open
  </button>
  <div
    class="modal fade show"
    style="display: block;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="modal-close-button btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component 1 5`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
    type="button"
  >
    Open
  </button>
  <div
    class="modal"
    style="display: none;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="modal-close-button btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component 1 6`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
    type="button"
  >
    Open
  </button>
  <div
    class="modal fade show"
    style="display: block;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="modal-close-button btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component 1 7`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
    type="button"
  >
    Open
  </button>
  <div
    class="modal"
    style="display: none;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="modal-close-button btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;
